lib/go-tc/servers.go

```diff
@@ -97,6 +99,30 @@ type ServerInterfaceInfo struct {
 	Name         string            `json:"name" db:"name"`
 }
 
+// GetDefaultAddress returns the IPv4 and IPv6 service addresses of the interface.
+func (i *ServerInterfaceInfo) GetDefaultAddress() (string, string) {
+	var ipv4 string
+	var ipv6 string
+	for _, ip := range i.IPAddresses {
+		if ip.ServiceAddress {
+			address, _, err := net.ParseCIDR(ip.Address)
+			if err != nil || address == nil {
+				continue
```

Comment:
nah. I generally consider importing the logger into the tc, utils, or rfc packages bad form, since they're just libraries, and should be able to run their code in situations where logging is impossible, not set up, or doesn't even make sense. So if anything, it should just return the error and make the caller deal with it.

But also, I didn't write this method. I just changed the receiver; we had two separate, equivalent definitions of server interfaces. The other one was in `traffic_router.go`. I deleted that one and moved this method to its new, more appropriate location. So I don't really wanna change it.

Reply:
+1 - libraries should return errors, never log. If this really needs to return a warning, the function can be documented to return a "non-fatal error", or "warnings string array" or something. But that doesn't seem valuable enough to me to warrant the awkwardness. I would vote documenting the behavior in the function doc, though, e.g. "Addresses which aren't valid CIDR are ignored." 

But +1 to everything @ocket8888 said.